avocado/core/test.py

```diff
@@ -312,21 +305,28 @@ def _get_tagged_name(self, logdir):
 
         :param logdir: Log directory being in use for result storage.
 
-        :return: Unique test name
+        :return: Unique test name and the logdir
         """
         name = self.name
         if self.tag is not None:
             name += ".%s" % self.tag
         tag = 0
         tagged_name = name
-        safe_tagged_name = astring.string_to_safe_path(tagged_name)
-        while os.path.isdir(os.path.join(logdir, safe_tagged_name)):
+        # The maximal length on ext4+python2.7 is 255 chars.
+        safe_tagged_name = astring.string_to_safe_path(tagged_name[:240])
+        for i in xrange(9999):
+            if not os.path.isdir(os.path.join(logdir, safe_tagged_name)):
+                break
             tag += 1
             tagged_name = "%s.%s" % (name, tag)
-            safe_tagged_name = astring.string_to_safe_path(tagged_name)
-
+            safe_tagged_name = astring.string_to_safe_path("%s.%s"
+                                                           % (name[:240], tag))
```

Comment:
yes I know, but I think it's unnecessarily complex. It's only suppose to be used when really long names are generated and getting few more chars didn't looked that important to introduce more potential bugs and higher cpu utilization (minor).

But if more people think it's essential to get as much characters as possible, I can use it.